crypto/err/err.c

```diff
@@ -790,7 +790,7 @@ void ERR_add_error_vdata(int num, va_list args)
     }
     len = strlen(str);
 
-    for (len = 0; --num >= 0; ) {
+    while (--num >= 0) {
```

Comment:
LGTM
The original code would have probably been cleaner to do in 2 passes - maybe as a function that returned a buffer (can be NULL) and return a *size.